app/assets/javascripts/cascade/new-home.js

```diff
@@ -1,1139 +1,1189 @@
+        processNavigation : function(input) {
+
+            if (cu_hero_area.isChanging) return false;
+            cu_hero_area.isChanging = true;
+
+            var max         = cu_hero_area.pastCampaigns.length -1; // zero based
+            var min         = 0;
+            var direction   = 'older'; // default transition
+            var current_num = cu_hero_area.currentCampaign;
+
+            if (input == 'newer') {
+                direction = 'newer';
+                current_num--;
+
+            } else if (input == 'older') {
+                direction = 'older';
+                current_num++;
+
+            } else if (typeof input === 'number' && input % 1 == 0) {
+                current_num = input;
+
+            }
+
+            if (current_num < min) return false;
+            if (current_num > max) return false;
+
+            cu_hero_area.currentCampaign = current_num;
+
+            if (cu_window_manager.useTransitions) {
+                $("#mastheadNavigation").fadeOut();
+            }
+
+            /* jshint -W069: ['foo'] is better written in dot notation. */
+            // Check for a local cached copy of this HTML
+            if (cu_hero_area.pastCampaigns[current_num]['html']) {
+                cu_hero_area.transitionToStory(cu_hero_area.pastCampaigns[current_num]['html'], direction);
+            } else {
+
+                var newFile = cu_hero_area.pastCampaigns[current_num]['filename'];
+
+                $.get(cu_hero_area.hero_stories_html_dir+newFile,function(data){
+                    cu_hero_area.pastCampaigns[current_num]['html'] = data;
+                    cu_hero_area.transitionToStory(data, direction);
+                });
+            }
+
+            // Update the URL
+            location.hash = "#story-"+cu_hero_area.pastCampaigns[current_num]['slug'];
+            /* jshint +W069 */
+
+        }, // end processNavigation
+
+        transitionToStory : function(raw_html, direction) {
+
+            $new_content = $(raw_html);
+            $old_content = $("#hero");
+
+            // Check for extra HTML wrap
+            if ($new_content.find("#hero").length > 0) {
+                $new_content = $new_content.find("#hero");
+            }
+
+
+            if (cu_window_manager.useTransitions) {
+
+                $new_content.css({
+                    'opacity' : 0
+                }).addClass('fast_transition');
+
+                $old_content.css({
+                    "opacity" : 1,
+                    "margin-top"    : "-"+$("#hero").outerHeight()+"px"
+                }).addClass('fast_transition');
+
+                $old_content.before($new_content);
+
+                // Need to wait until the transition class is in the DOM
+                setTimeout(function() {
+                    $new_content.css({
+                        "opacity":1
+                    });
+
+                    $old_content.css({
+                     
```

Comment:
I rewrote this module. It used to be called `heroQuickView`. I'm not convinced it ever worked. Also ran jshint on the entire file and fixed most the errors.